### ISSUE TITLE
package linux-tools for zabbly kernel?

### ISSUE THREAD

**Issue by mtippmann** [zabbly/linux#2]
thanks for creating this! 

I have a minor suggestion: tools like `cpupower` and `x86_energy_performance` are packaged in ubuntu in kernel-specific packages `linux-tools-<kernel-variant>-<version>` - would be great if the corresponding packages would also exist for zabbly kernel.

**Comment by stgraber** [comment]
The packages come from the `mkdebian` script that's part of the mainline Linux kernel.
That logic doesn't not currently generate packages for some of those tools that are provided within the kernel tree.

I think it could be interesting to have those, but rather than have the Zabbly packaging add a bunch of packaging logic on top of what's provided by the kernel tree, this should probably be contributed directly to `mkdebian` in the upstream tree.

**Comment by 3v1n0** [comment]
Oh, I was trying to get `perf` working in latest kernels, but that's somewhat a missing piece. Any way to get this?

**Comment by mtippmann** [comment]
> Oh, I was trying to get `perf` working in latest kernels, but that's somewhat a missing piece. Any way to get this?

my workaround is just setting symlinks and install latest tools - for most use cases they don't need a hard dependency on the kernel 

```
user@machine:/usr/lib/linux-tools$ ls -al
insgesamt 28
drwxr-xr-x   4 root root   7 Nov 22 06:36 .
drwxr-xr-x 116 root root 200 Nov 22 06:39 ..
drwxr-xr-x   2 root root   3 Nov 22 06:39 5.15.0-88-generic
drwxr-xr-x   2 root root  11 Nov 22 06:36 5.15.0-89-generic
lrwxrwxrwx   1 root root  18 Nov  4 01:57 6.5.6-zabbly+ -> 5.15.0-88-generic/
lrwxrwxrwx   1 root root  18 Nov  4 01:58 6.5.8-zabbly+ -> 5.15.0-88-generic/
lrwxrwxrwx   1 root root  18 Nov  4 01:58 6.5.9-zabbly+ -> 5.15.0-88-generic/
```

I failed at my last attempt to refresh the links on every upgrade  but with enough `dpkg` foo this should be possible.

**Comment by stgraber** [comment]
Yeah, I still think the best option will be to introduce a new binary package to the kernel build, similar to how linux-libc-dev is handled. It shouldn't be too difficult to do, though not having it conflict with how Debian and Ubuntu do things may be slightly more challenging (as the kernel packaging logic is mostly distro agnostic).

**Comment by graveland** [comment]
In order to get tools like `perf`, we'd have to clone/check out the exact source that a particular zabbly kernel was built from, and build our own binaries and packages, then distribute the resulting package to all of our systems, correct? If we even want to go down that path, what is the way we'd determine the exact commit hash of a particular kernel?

Has there been any movement or work relating to this?

**Comment by stgraber** [comment]
You should be able to enable source packages for the Linux kernel repo, then `apt-get source` will let you get the source tarball for the kernel.

**Comment by graveland** [comment]
Yeah, that's what I'm going to do. Ideally the tools should be built with the same sources as the kernel, but those are only available for a very short period for zabbly. I don't suppose they're archived anywhere instead of just deleted once they're older than whatever your retention period is?

**Comment by stgraber** [comment]
Ah yeah, the kernel source packages get removed from the repo at the same time as the binary packages and I only keep a few around as each kernel build takes quite a lot of space.

**Comment by graveland** [comment]
I guess this is an argument for building and publishing tools along with the kernel? :)
